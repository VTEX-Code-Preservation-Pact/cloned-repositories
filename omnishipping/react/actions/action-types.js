export const CHANGE_ACTIVE_TAB = 'CHANGE_ACTIVE_TAB'
export const CHANGE_ACTIVE_SLA_OPTION = 'CHANGE_ACTIVE_SLA_OPTION'
export const SELECT_DELIVERY_WINDOW = 'SELECT_DELIVERY_WINDOW'

export const CHANGE_ACTIVE_PICKUP_OPTIONS = 'CHANGE_ACTIVE_PICKUP_OPTIONS'
export const CHANGE_ACTIVE_PICKUP_SELLER = 'CHANGE_ACTIVE_PICKUP_SELLER'
export const CHANGE_ACTIVE_DELIVERY_CHANNEL = 'CHANGE_ACTIVE_DELIVERY_CHANNEL'
export const CLOSE_PICKUP_MODAL = 'CLOSE_PICKUP_MODAL'
export const OPEN_PICKUP_MODAL = 'OPEN_PICKUP_MODAL'

export const CHANGE_ACTIVE_PICKUP_DETAILS = 'CHANGE_ACTIVE_PICKUP_DETAILS'
export const CLOSE_PICKUP_DETAILS = 'CLOSE_PICKUP_DETAILS'

export const UPDATE_ADDRESS_FORM = 'UPDATE_ADDRESS_FORM'
export const UPDATE_ADDRESS_RULES = 'UPDATE_ADDRESS_RULES'
export const CHANGE_SELECTED_ADDRESS_RULES = 'CHANGE_SELECTED_ADDRESS_RULES'
export const VALIDATE_ADDRESS_FORM = 'VALIDATE_ADDRESS_FORM'
export const VALIDATE_ADDRESS_FORM_FIELDS = 'VALIDATE_ADDRESS_FORM_FIELDS'
export const VALIDATE_POSTAL_CODE = 'VALIDATE_POSTAL_CODE'
export const ADD_ADDRESS_FORM_ADDRESS = 'ADD_ADDRESS_FORM_ADDRESS'
export const ADD_DELIVERY_ADDRESS = 'ADD_DELIVERY_ADDRESS'
export const UPDATE_ORDERFORM = 'UPDATE_ORDERFORM'

export const SELECT_ADDRESS_ITEM = 'SELECT_ADDRESS_ITEM'

export const EDIT_ADDRESS = 'EDIT_ADDRESS'
export const EDIT_OMNISHIPPING = 'EDIT_OMNISHIPPING'
export const NEW_ADDRESS = 'NEW_ADDRESS'
export const SHOW_ADDRESS_PRODUCTS = 'SHOW_ADDRESS_PRODUCTS'
export const SHOW_ADDRESS = 'SHOW_ADDRESS'
export const SHOW_SPLIT = 'SHOW_SPLIT'
export const SHOW_ADDRESS_LIST = 'SHOW_ADDRESS_LIST'
export const SHOW_PAYMENT_BUTTON = 'SHOW_PAYMENT_BUTTON'
export const SHOW_OMNISHIPPING_COMPLETED = 'SHOW_OMNISHIPPING_COMPLETED'
export const SHOW_OMNISHIPPING = 'SHOW_OMNISHIPPING'
export const ENABLE_OMNISHIPPING = 'ENABLE_OMNISHIPPING'
export const DISABLE_OMNISHIPPING = 'DISABLE_OMNISHIPPING'

export const UPDATE_SHIPPING_FAILURE = 'UPDATE_SHIPPING_FAILURE'
export const UPDATE_SHIPPING_FROM_SHIPPING_RATE_PREVIEW =
  'UPDATE_SHIPPING_FROM_SHIPPING_RATE_PREVIEW'
export const UPDATE_SHIPPING_REQUEST = 'UPDATE_SHIPPING_REQUEST'

export const SIMULATE_SHIPPING_REQUEST = 'SIMULATE_SHIPPING_REQUEST'
export const SIMULATE_SHIPPING_FAILURE = 'SIMULATE_SHIPPING_FAILURE'
export const SIMULATE_SHIPPING_SUCCESS = 'SIMULATE_SHIPPING_SUCCESS'

export const UPDATE_SIMULATION_OPTIONS = 'UPDATE_SIMULATION_OPTIONS'

export const REMOVE_ITEMS_REQUEST = 'REMOVE_ITEMS_REQUEST'

export const FOCUS_ON_INPUT = 'FOCUS_ON_INPUT'

export const HAS_PICKUP_OPTIONS_EVENT = 'HAS_PICKUP_OPTIONS_EVENT'
export const SELECTED_PICKUP_TAB_EVENT = 'SELECTED_PICKUP_TAB_EVENT'
export const SHIPPING_STATS_EVENT = 'SHIPPING_STATS_EVENT'
export const SEARCHED_PICKUP_ADDRESS_EVENT = 'SEARCHED_PICKUP_ADDRESS_EVENT'
export const SEARCHED_PICKUP_ADDRESS_GEOLOCATION_EVENT =
  'SEARCHED_PICKUP_ADDRESS_GEOLOCATION_EVENT'

export const VTEX_ID_STARTED = 'VTEX_ID_STARTED'
export const VTEX_ID_CLOSED = 'VTEX_ID_CLOSED'

export const CHANGE_ACTIVE_DELIVERY_PACKAGE = 'CHANGE_ACTIVE_DELIVERY_PACKAGE'

export const UPDATE_SELECTED_PACKAGE_SLIDER = 'UPDATE_SELECTED_PACKAGE_SLIDER'

export const HAS_GEOLOCATION = 'HAS_GEOLOCATION'

export const SELECT_SCHEDULED_DELIVERY = 'SELECT_SCHEDULED_DELIVERY'
export const TOGGLE_SCHEDULED_DELIVERY = 'TOGGLE_SCHEDULED_DELIVERY'

export const OMNISHIPPING_JS_ERROR = 'OMNISHIPPING_JS_ERROR'
export const OMNISHIPPING_API_ERROR = 'OMNISHIPPING_API_ERROR'

export const SET_ADITIONAL_SHIPPING_DATA = 'SET_ADITIONAL_SHIPPING_DATA'

export const OMNISHIPPING_PAGEVIEW_EVENT = 'OMNISHIPPING_PAGEVIEW_EVENT'

export const COMPLETE_OMNISHIPPING = 'COMPLETE_OMNISHIPPING'
