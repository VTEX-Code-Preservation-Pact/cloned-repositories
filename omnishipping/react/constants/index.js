export const PICKUP_IN_STORE = 'pickup-in-point'
export const DELIVERY = 'delivery'
export const PICKUP = 'pickup'
export const SCHEDULED_DELIVERY = 'Entrega Agendada'

export const RESIDENTIAL = 'residential'
export const COMMERCIAL = 'commercial'
export const SEARCH = 'search'
export const GIFT_REGISTRY = 'giftRegistry'

export const GEOCORDS = 'geoCoords'
export const ACCOUNT_INFO_ENDOPOINT =
  '/api/logistics/pub/shipping/configuration'
export const HIDE_MAP = 'HIDE_MAP'
export const SHOW_MAP = 'SHOW_MAP'
export const AVAILABLE = 'available'
export const SHIPPING_HASH = '#/shipping'
export const PAYMENT_HASH = '#/payment'

export const CHEAPEST = 'CHEAPEST'
export const FASTEST = 'FASTEST'
export const COMBINED = 'COMBINED'

export const DENIED = 'denied'

export const VTEXLOCAL = 'vtexlocal'
export const VTEXBETA = 'vtexcommercebeta'
export const VTEXSTABLE = 'vtexcommercestable'

export const DEFAULT_WIDTH = 50
export const DEFAULT_HEIGHT = 50

export const IMPORTANT = 'Important'
export const CRITICAL = 'Critical'
export const ERROR = 'Error'
export const INFO = 'Info'

export const OMNISHIPPING = 'omnishipping'
