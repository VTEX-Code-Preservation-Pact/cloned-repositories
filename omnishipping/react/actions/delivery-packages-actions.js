import * as types from './action-types'

export const updateSelectedPackageSlider = packageIndex => ({
  type: types.UPDATE_SELECTED_PACKAGE_SLIDER,
  packageIndex,
})
