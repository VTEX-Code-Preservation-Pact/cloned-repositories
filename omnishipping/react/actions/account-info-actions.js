import * as types from './action-types'
import {
  getAccountName,
  getLocale,
  getUserEmail,
} from '../selectors/order-form-selectors'
import { AUTHENTICATED_USER } from '../constants/checkout-events'
const $ = window.$

export const vtexIdStarted = () => ({
  type: types.VTEX_ID_STARTED,
})

export const closedVtexId = () => ({
  type: types.VTEX_ID_CLOSED,
})

export const startVtexID = callback => (dispatch, getState) => {
  const state = getState()

  dispatch(vtexIdStarted())

  window.vtexid &&
    window.vtexid.start({
      returnUrl: window.location.href,
      accountName: getAccountName(state) || window.vtex.accountName,
      locale: getLocale(state),
      email: getUserEmail(state),
    })

  const authCallback = () => callback && callback(true)
  if ($) {
    $(window).one(AUTHENTICATED_USER, authCallback)
    $(window).one('closed.vtexid', () => {
      dispatch(closedVtexId())
      $(window).off(AUTHENTICATED_USER, authCallback)
    })
  }
}
