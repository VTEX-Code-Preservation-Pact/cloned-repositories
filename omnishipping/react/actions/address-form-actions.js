import * as types from './action-types'
import { newAddress } from '../utils/AddressFormUtils'
import {
  getDefaultShipsTo,
  getStoreCountryCode,
} from '../selectors/order-form-selectors'
import { getSearchAddress } from '../selectors/address-form-selectors'
import { newAddressEvent } from './component-activity-actions'
import { SEARCH } from '../constants'
import { helpers } from 'vtex.address-form'
const { removeValidation } = helpers

export const updateAddressForm = ({
  address,
  searchAddress,
  channel,
  rules,
}) => ({
  type: types.UPDATE_ADDRESS_FORM,
  address,
  searchAddress,
  channel,
  rules,
})

export const validateAddressForm = rules => ({
  type: types.VALIDATE_ADDRESS_FORM,
  rules,
})

export const validateAddressFormFields = rules => ({
  type: types.VALIDATE_ADDRESS_FORM_FIELDS,
  rules,
})

export const validatePostalCode = rules => ({
  type: types.VALIDATE_POSTAL_CODE,
  rules,
})

export const addAddressFormAddress = ({ address, searchAddress, channel }) => ({
  type: types.ADD_ADDRESS_FORM_ADDRESS,
  address,
  searchAddress,
  channel,
})

export const focusOnInput = input => ({
  type: types.FOCUS_ON_INPUT,
  input,
})

export const addDeliveryAddress = () => (dispatch, getState) => {
  dispatch(newAddressEvent(true))
  const state = getState()
  const shipsTo = getDefaultShipsTo(state)
  const storeCountryCode = getStoreCountryCode(state)
  const channel = state.componentActivity.activeDeliveryChannel
  const searchAddress = removeValidation(getSearchAddress(state))
  const country =
    shipsTo.indexOf(storeCountryCode) > -1 ? storeCountryCode : shipsTo[0]
  dispatch(
    addAddressFormAddress({
      address: newAddress({ country }),
      searchAddress,
      channel,
    })
  )
}

export const addNewAddresses = () => (dispatch, getState) => {
  const state = getState()
  const shipsTo = getDefaultShipsTo(state)
  const storeCountryCode = getStoreCountryCode(state)
  const channel = state.componentActivity.activeDeliveryChannel
  const country =
    shipsTo.indexOf(storeCountryCode) > -1 ? storeCountryCode : shipsTo[0]
  dispatch(
    addAddressFormAddress({
      address: newAddress({ country }),
      searchAddress: newAddress({
        country,
        addressType: SEARCH,
      }),
      channel,
    })
  )
}
