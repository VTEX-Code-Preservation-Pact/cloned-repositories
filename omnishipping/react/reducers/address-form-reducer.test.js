import reducer from './address-form-reducer'
import * as types from '../actions/action-types'
import { addValidation } from '@vtex/address-form'
import BRA from '@vtex/address-form/lib/country/BRA'
import MockDate from 'mockdate'
import { DELIVERY } from '../constants/index'
const defaultAddress = {
  city: 'Rio de Janeiro',
  neighborhood: 'Botafogo',
  receiverName: 'Xablauzera da Bahia',
  addressId: '11',
  addressType: 'residential',
  complement: null,
  country: 'BRA',
  geoCoordinates: [],
  number: '12',
  postalCode: '22071-060',
  reference: null,
  state: 'Rio de Janeiro',
  street: 'Rua Saint Roman',
  addressQuery: '',
}

describe('AddressForm reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle UPDATE_ORDERFORM', () => {
    MockDate.set(Date.UTC(2017, 6, 14, 18, 0, 0, 0), 180)
    const orderForm = {
      storePreferencesData: {
        countryCode: 'BRA',
      },
      shippingData: {
        selectedAddresses: [
          {
            ...defaultAddress,
            neighborhood: 'Flamengo',
          },
        ],
      },
    }

    expect(
      reducer(
        {
          residentialId: defaultAddress.addressId,
          addresses: {
            [defaultAddress.addressId]: addValidation(defaultAddress),
          },
        },
        {
          type: types.UPDATE_ORDERFORM,
          orderForm,
        }
      )
    ).toMatchSnapshot()
  })

  it('should handle UPDATE_ORDERFORM with different addressId', () => {
    MockDate.set(Date.UTC(2017, 6, 14, 18, 0, 0, 0), 180)
    const orderForm = {
      storePreferencesData: {
        countryCode: 'BRA',
      },
      canEditData: true,
      shippingData: {
        selectedAddresses: [
          {
            ...defaultAddress,
            addressId: '1234',
            neighborhood: 'Flamengo',
          },
        ],
      },
    }

    expect(
      reducer(
        {
          residentialId: defaultAddress.addressId,
          valid: false,
          addresses: {
            [defaultAddress.addressId]: addValidation(defaultAddress),
          },
        },
        { type: types.UPDATE_ORDERFORM, orderForm, channel: DELIVERY }
      )
    ).toMatchSnapshot()
  })

  it('should handle UPDATE_ADDRESS_FORM', () => {
    const searchId = '12'
    const defaultState = {
      residentialId: defaultAddress.addressId,
      searchId: searchId,
      addresses: {
        [defaultAddress.addressId]: addValidation(defaultAddress),
        [searchId]: addValidation({ ...defaultAddress, addressId: searchId }),
      },
    }

    const address = addValidation({
      ...defaultAddress,
      city: 'Rio de Janeiro',
      complement: 'Apto 106',
      number: '10',
    })

    expect(
      reducer(defaultState, {
        type: types.UPDATE_ADDRESS_FORM,
        address,
        searchAddress: defaultState.addresses[defaultState.searchId],
      })
    ).toMatchSnapshot()
  })

  it('should handle UPDATE_ADDRESS_FORM sending only searchAddress', () => {
    const searchId = '12'
    const defaultState = {
      residentialId: defaultAddress.addressId,
      searchId: searchId,
      addresses: {
        [defaultAddress.addressId]: addValidation(defaultAddress),
        [searchId]: addValidation({ ...defaultAddress, addressId: searchId }),
      },
    }

    const searchAddress = addValidation({
      ...defaultAddress,
      city: 'Rio de Janeiro',
      complement: 'Apto 106',
      number: '10',
    })

    expect(
      reducer(defaultState, {
        type: types.UPDATE_ADDRESS_FORM,
        searchAddress,
      })
    ).toMatchSnapshot()
  })

  it('should handle VALIDATE_ADDRESS_FORM', () => {
    const defaultState = {
      residentialId: defaultAddress.addressId,
      addresses: {
        [defaultAddress.addressId]: addValidation({
          ...defaultAddress,
          city: 'Rio de Janeiro',
          neighborhood: 'Gávea',
          receiverName: 'Teste',
        }),
      },
    }

    const rules = BRA

    expect(
      reducer(defaultState, {
        type: types.VALIDATE_ADDRESS_FORM,
        rules,
      })
    ).toMatchSnapshot()
  })

  it('should not break with rules=null', () => {
    const defaultState = {
      residentialId: defaultAddress.addressId,
      addresses: {
        [defaultAddress.addressId]: addValidation({
          ...defaultAddress,
          city: 'Rio de Janeiro',
          neighborhood: 'Gávea',
          receiverName: 'Teste',
        }),
      },
      queuedValidations: [],
    }

    const rules = null

    expect(
      reducer(defaultState, {
        type: types.VALIDATE_ADDRESS_FORM,
        rules,
      })
    ).toMatchSnapshot()
  })

  it('should handle VALIDATE_ADDRESS_FORM_FIELDS', () => {
    const defaultState = {
      residentialId: defaultAddress.addressId,
      addresses: {
        [defaultAddress.addressId]: addValidation({
          ...defaultAddress,
          city: 'Rio de Janeiro',
          neighborhood: 'Gávea',
          receiverName: 'Teste',
        }),
      },
    }

    const rules = BRA

    expect(
      reducer(defaultState, {
        type: types.VALIDATE_ADDRESS_FORM_FIELDS,
        rules,
      })
    ).toMatchSnapshot()
  })

  it('should handle VALIDATE_POSTAL_CODE', () => {
    const defaultState = {
      addresses: {
        [defaultAddress.addressId]: addValidation({
          ...defaultAddress,
          postalCode: '22071-060',
        }),
      },
      residentialId: defaultAddress.addressId,
    }

    const rules = BRA

    expect(
      reducer(defaultState, {
        type: types.VALIDATE_POSTAL_CODE,
        rules,
      })
    ).toMatchSnapshot()
  })

  it('should handle ADD_DELIVERY_ADDRESS', () => {
    const defaultState = {
      [defaultAddress.addressId]: addValidation(defaultAddress),
    }

    const address = addValidation({
      ...defaultAddress,
      city: 'São Paulo',
      neighborhood: 'Jardim Botânico',
      receiverName: 'Bananinha',
    })

    expect(
      reducer(defaultState, {
        type: types.ADD_DELIVERY_ADDRESS,
        address,
      })
    ).toMatchSnapshot()
  })

  it('should handle FOCUS_ON_INPUT', () => {
    const input = {
      name: 'postalCode',
      value: '',
    }
    const defaultState = {
      addresses: {
        [defaultAddress.addressId]: addValidation(defaultAddress),
      },
      residentialId: defaultAddress.addressId,
    }
    const action = {
      type: types.FOCUS_ON_INPUT,
      input,
    }
    const resultState = reducer(defaultState, action)

    expect(
      resultState.addresses[defaultAddress.addressId][input.name].focus
    ).toEqual(true)
    expect(
      resultState.addresses[defaultAddress.addressId][input.name].value
    ).toEqual('')
  })
})
