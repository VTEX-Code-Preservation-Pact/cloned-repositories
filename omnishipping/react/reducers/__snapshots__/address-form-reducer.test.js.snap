// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm reducer should handle ADD_DELIVERY_ADDRESS 1`] = `
Object {
  "11": Object {
    "addressId": Object {
      "value": "11",
    },
    "addressQuery": Object {
      "value": "",
    },
    "addressType": Object {
      "value": "residential",
    },
    "city": Object {
      "value": "Rio de Janeiro",
    },
    "complement": Object {
      "value": null,
    },
    "country": Object {
      "value": "BRA",
    },
    "geoCoordinates": Object {
      "value": Array [],
    },
    "neighborhood": Object {
      "value": "Botafogo",
    },
    "number": Object {
      "value": "12",
    },
    "postalCode": Object {
      "value": "22071-060",
    },
    "receiverName": Object {
      "value": "Xablauzera da Bahia",
    },
    "reference": Object {
      "value": null,
    },
    "state": Object {
      "value": "Rio de Janeiro",
    },
    "street": Object {
      "value": "Rua Saint Roman",
    },
  },
}
`;

exports[`AddressForm reducer should handle UPDATE_ADDRESS_FORM 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": "Apto 106",
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Botafogo",
      },
      "number": Object {
        "value": "10",
      },
      "postalCode": Object {
        "loading": undefined,
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
    "12": Object {
      "addressId": Object {
        "value": "12",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "search",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Botafogo",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "alreadyCompletedReceiverName": false,
  "residentialId": "11",
  "searchId": "12",
}
`;

exports[`AddressForm reducer should handle UPDATE_ADDRESS_FORM sending only searchAddress 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Botafogo",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "loading": undefined,
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
    "12": Object {
      "addressId": Object {
        "value": "12",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "search",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": "Apto 106",
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Botafogo",
      },
      "number": Object {
        "value": "10",
      },
      "postalCode": Object {
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "alreadyCompletedReceiverName": false,
  "residentialId": "11",
  "searchId": "12",
}
`;

exports[`AddressForm reducer should handle UPDATE_ORDERFORM 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Botafogo",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "loading": false,
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "residentialId": "11",
}
`;

exports[`AddressForm reducer should handle UPDATE_ORDERFORM with different addressId 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Botafogo",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
    "1234": Object {
      "addressId": Object {
        "value": "1234",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "postalCodeAutoCompleted": true,
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "postalCodeAutoCompleted": true,
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "postalCodeAutoCompleted": true,
        "value": "Flamengo",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "postalCodeAutoCompleted": true,
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "postalCodeAutoCompleted": true,
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "postalCodeAutoCompleted": true,
        "value": "Rua Saint Roman",
      },
    },
    "1500055200000": Object {
      "addressId": Object {
        "value": "1500055200000",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "search",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Flamengo",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "receiverName": undefined,
  "residentialId": "1234",
  "searchId": "1500055200000",
  "valid": false,
}
`;

exports[`AddressForm reducer should handle VALIDATE_ADDRESS_FORM 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Gávea",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Teste",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "queuedValidations": Array [],
  "residentialId": "11",
  "valid": false,
}
`;

exports[`AddressForm reducer should handle VALIDATE_ADDRESS_FORM_FIELDS 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Gávea",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Teste",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "focus": true,
        "reason": "ERROR_VALUE_IS_NOT_AN_OPTION",
        "valid": false,
        "value": "Rio de Janeiro",
        "valueOptions": undefined,
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "queuedValidations": Array [],
  "residentialId": "11",
  "valid": false,
}
`;

exports[`AddressForm reducer should handle VALIDATE_POSTAL_CODE 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Botafogo",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "reason": undefined,
        "valid": true,
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Xablauzera da Bahia",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "queuedValidations": Array [],
  "residentialId": "11",
}
`;

exports[`AddressForm reducer should not break with rules=null 1`] = `
Object {
  "addresses": Object {
    "11": Object {
      "addressId": Object {
        "value": "11",
      },
      "addressQuery": Object {
        "value": "",
      },
      "addressType": Object {
        "value": "residential",
      },
      "city": Object {
        "value": "Rio de Janeiro",
      },
      "complement": Object {
        "value": null,
      },
      "country": Object {
        "value": "BRA",
      },
      "geoCoordinates": Object {
        "value": Array [],
      },
      "neighborhood": Object {
        "value": "Gávea",
      },
      "number": Object {
        "value": "12",
      },
      "postalCode": Object {
        "value": "22071-060",
      },
      "receiverName": Object {
        "value": "Teste",
      },
      "reference": Object {
        "value": null,
      },
      "state": Object {
        "value": "Rio de Janeiro",
      },
      "street": Object {
        "value": "Rua Saint Roman",
      },
    },
  },
  "queuedValidations": Array [
    "VALIDATE_ADDRESS_FORM",
  ],
  "residentialId": "11",
}
`;

exports[`AddressForm reducer should return the initial state 1`] = `
Object {
  "addresses": Object {},
  "alreadyCompletedReceiverName": false,
  "queuedValidations": Array [],
  "receiverName": null,
  "valid": null,
}
`;
